source/src/microapp/cs_MicroappController.cpp

```diff
@@ -579,21 +620,45 @@ bool MicroappController::registerSoftInterruptSlotBle(uint8_t id) {
 	for (int i = 0; i < MICROAPP_MAX_BLE_ISR_COUNT; ++i) {
 		if (!_bleIsr[i].registered) {
 			_bleIsr[i].registered = true;
-			_bleIsr[i].type       = BleEventDeviceScanned;
+			_bleIsr[i].type       = BleEventDeviceScanned; // TODO: change if more types
 			_bleIsr[i].id         = id;
-			LOGi("Registered slot %i for BLE events towards the microapp", id);
+			LOGi("Registered slot %i for BLE events towards the microapp (id=%d)", i, id);
 			return true;
 		}
 	}
 	// no empty slot available
-	LOGw("Could not register slot for BLE events for the microapp %i", id);
+	LOGw("Could not register slot for BLE events for the microapp (id=%i)", id);
 	return false;
 }
 
+/*
+ * Register the slot for a Mesh soft interrupt.
+ */
+bool MicroappController::registerSoftInterruptSlotMesh(uint8_t id) {
+	if (!_meshIsr.registered) {
+		LOGi("Registered the softInterrupt slot for mesh interrupts (id=%d)", id);
+		_meshIsr.registered = true;
+		_meshIsr.id = id;
+		return true;
+	}
+	else { // already registered
+		LOGw("Could not register slot for Mesh events for the microapp (id=%i)", id);
+		return false;
+	}
+}
+
 void MicroappController::setScanning(bool scanning) {
 	_microappIsScanning = scanning;
 }
 
+void MicroappController::setEmptySoftInterrupts(uint8_t emptySoftInterrupts) {
+	_emptySoftInterrupts = emptySoftInterrupts;
+}
+
+void MicroappController::incrementEmptySoftInterrupts() {
+	_emptySoftInterrupts++;
```

Comment:
There are no checks here.

Reply:
What kind of checks? The only one I can imagine is checking if you're not going over the maximum number of soft interrupt slots. However, as explained in a comment somewhere else in this PR right now bluenet is agnostic towards the maximum interrupt slots since it's only defined on the microapp side. If that is changed then a check would make sense.

At the very least I can make sure the uint8_t won't overflow.